glib/settings_schema_source.go

```diff
@@ -50,15 +53,33 @@ func SettingsSchemaSourceNewFromDirectory(dir string, parent *SettingsSchemaSour
 	cstr := (*C.gchar)(C.CString(dir))
 	defer C.free(unsafe.Pointer(cstr))
 
-	return wrapSettingsSchemaSource(C.g_settings_schema_source_new_from_directory(cstr, parent.native(), gbool(trusted), nil))
+	schemaSource := C.g_settings_schema_source_new_from_directory(cstr, parent.native(), gbool(trusted), nil)
+	if schemaSource == nil {
+		return nil
+	}
+
+	//The finalizer is on the glib.SettingsSchemaSource.
+	s := wrapSettingsSchemaSource(schemaSource)
+	runtime.SetFinalizer(s, (*SettingsSchemaSource).Unref)
```

Comment:
Who calls `Ref()` here? simply adding `Unref()` may be bad.

Reply:
true that. I'll add a call of Ref()

Reply:
The func sets the ref count to 1. So we can leave it as-is

https://github.com/GNOME/glib/blob/3499bd7713adcb28b020d36bf06eda984fc26da6/gio/gsettingsschema.c#L311